### PR TITLE
2023 01 17 extern version

### PR DESCRIPTION
**Pull request by thedavidmeister**
this introduces `__hc__` prefix for imports and exports and also `_N` version suffix for both

arguably less useful for exports, but this paves the way for a host to inspect the guest via tooling, e.g. `module.imports()` and then interact with it different, e.g. for backwards compatibility, based on the version of the function that is provided/expected

### CODE REVIEW

**Review by neonphog — APPROVED** [review]
Awesome :+1:

File: test/wasm_memory/src/wasm.rs [hunk 1]
```diff
@@ -1,17 +1,17 @@
 use holochain_wasmer_guest::*;
 
 host_externs!(
-    __debug
+    debug:1
```
Comment (jost-s) [inline test/wasm_memory/src/wasm.rs:4]: Why is this changed?

**Review by jost-s — APPROVED** [review]

File: test/wasm_memory/src/wasm.rs (hunk 1 above)
Comment (jost-s) [reply test/wasm_memory/src/wasm.rs:4]: Sweet, thanks
